GW2EIEvtcParser/ParsedData/CombatData.cs

```diff
@@ -1,1746 +1,1409 @@
+    {
+        return _breakbarDamageTakenData.GetValueOrEmpty(dst);
+    }
+
+    /// <summary>
+    /// Returns list of crowd control events taken by Agent
+    /// </summary>
+    public IReadOnlyList<CrowdControlEvent> GetIncomingCrowdControlData(AgentItem dst)
+    {
+        return _crowControlTakenData.GetValueOrEmpty(dst);
+    }
+
+    public IReadOnlyList<AbstractMovementEvent> GetMovementData(AgentItem src)
+    {
+        return _statusEvents.MovementEvents.GetValueOrEmpty(src);
+    }
+
+    public IReadOnlyList<EffectEvent> GetEffectEventsBySrc(AgentItem src)
+    {
+        return _statusEvents.EffectEventsBySrc.GetValueOrEmpty(src);
+    }
 
-        /// <summary>
-        /// Returns list of cast events from skill
-        /// </summary>
-        /// <param name="skillID"></param> ID of the skill
-        /// <returns></returns>
-        public IReadOnlyList<AnimatedCastEvent> GetAnimatedCastData(long skillID)
-        {
-            if (_animatedCastDataById.TryGetValue(skillID, out List<AnimatedCastEvent> res))
-            {
-                return res;
-            }
-            return new List<AnimatedCastEvent>();
-        }
+    public IReadOnlyList<EffectEvent> GetEffectEventsByDst(AgentItem dst)
+    {
+        return _statusEvents.EffectEventsByDst.GetValueOrEmpty(dst);
+    }
 
-        /// <summary>
-        /// Returns list of damage events taken by Agent
-        /// </summary>
-        /// <param name="dst"></param> Agent
-        /// <returns></returns>
-        public IReadOnlyList<AbstractHealthDamageEvent> GetDamageTakenData(AgentItem dst)
-        {
-            if (_damageTakenData.TryGetValue(dst, out List<AbstractHealthDamageEvent> res))
-            {
-                return res;
-            }
-            return new List<AbstractHealthDamageEvent>();
-        }
+    public IReadOnlyList<EffectEvent> GetEffectEventsByEffectID(long effectID)
+    {
+        return _statusEvents.EffectEventsByEffectID.GetValueOrEmpty(effectID);
+    }
 
-        /// <summary>
-        /// Returns list of breakbar damage events taken by Agent
-        /// </summary>
-        /// <param name="dst"></param> Agent
-        /// <returns></returns>
-        public IReadOnlyList<BreakbarDamageEvent> GetBreakbarDamageTakenData(AgentItem dst)
+    /// <returns>true on success</returns>
+    public bool TryGetEffectEventsByGUID(GUID effectGUID, [NotNullWhen(true)] out IReadOnlyList<EffectEvent>? effectEvents)
+    {
+        var effectGUIDEvent = GetEffectGUIDEvent(effectGUID);
+        if (effectGUIDEvent != null)
         {
-            if (_breakbarDamageTakenData.TryGetValue(dst, out List<BreakbarDamageEvent> res))
+            effectEvents = GetEffectEventsByEffectID(effectGUIDEvent.ContentID);
+            if (effectEvents.Count > 0)
             {
-                return res;
+                return true;
             }
-            return new List<BreakbarDamageEvent>();
         }
```

Comment:
Why the inconsistent behavior on the "out" of TryGets? Some put null when nothing is found, others a empty list.

Reply:
Assuming you are talking about the TryGetEffectEventsByGUID function; this did not change from the original implementation, which is here: https://github.com/baaron4/GW2-Elite-Insights-Parser/pull/933/files/8c9b50cc509cebc9413ebbdafee850160fe0c526#diff-d5e44689367f787448b7f0dfe2eee8beb2da1421aabd9d2e78d663e33736ba32L1306-L1320 

and also returns null on failure.

Reply:
I don't have a strong opinion on wether it should be null or an empty list but I would prefer the behavior to be consistent. If your preference is "we have already created the list, just put it on the out", that is perfectly acceptable.

Reply:
Yea but this is exactly why they behave differently in the first place: one of them creates the list no matter what, and the other one filters an enumerable to obtain the result. 

Maybe some renaming in in order here, because "success" doesn't even make sense in the context of merging multiple lists. I will look into just dropping the "Try" prefix from some of these and see if that cleans it up.